File: stage1/settings-plugins/src/main/java/net/rubygrapefruit/plugins/stage1/SettingsPlugin.java
package net.rubygrapefruit.plugins.stage1;

import net.rubygrapefruit.plugins.stage0.BuildConstants;
import org.gradle.api.Plugin;
import org.gradle.api.initialization.Settings;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("unused")
public class SettingsPlugin implements Plugin<Settings> {
    @Override
    public void apply(@NotNull Settings target) {
        target.getPlugins().apply(BuildConstants.constants.foojay.plugin.id);

        target.getGradle().rootProject(project -> {
            project.getBuildscript().getRepositories().mavenCentral();
            project.getBuildscript().getDependencies().add("classpath", BuildConstants.constants.kotlin.plugin.coordinates);
            project.getBuildscript().getDependencies().add("classpath", BuildConstants.constants.serialization.plugin.coordinates);
        });
    }
}

File: bootstrap-plugins/src/main/kotlin/net/rubygrapefruit/plugins/bootstrap/SettingsPlugin.kt
package net.rubygrapefruit.plugins.bootstrap

import org.gradle.api.Plugin
import org.gradle.api.initialization.Settings
import net.rubygrapefruit.plugins.app.Versions

class SettingsPlugin : Plugin<Settings> {
    override fun apply(target: Settings) {
        target.run {
            plugins.apply("org.gradle.toolchains.foojay-resolver-convention")
            gradle.rootProject { project ->
                project.run {
                    buildscript.repositories.mavenCentral()
                    buildscript.dependencies.add("classpath", Versions.kotlin.pluginCoordinates)
                }
            }
        }
    }
}